ivadomed/losses.py

```diff
@@ -114,3 +114,103 @@ def forward(self, input, target):
         denominator = ((input + target).sum(-1) * class_weights).sum()
 
         return 1. - 2. * intersect / denominator.clamp(min=self.epsilon)
+
+
+class TverskyLoss(nn.Module):
+    """Tversky Loss.
+
+    Compute the Tversky loss defined in:
+        Sadegh et al. (2017) Tversky loss function for image segmentation using 3D fully convolutional deep networks
+
+    """
+
+    def __init__(self, alpha=0.3, beta=0.7, smooth=1.0):
+        """
+        Args:
+            alpha (float): weight of false positive voxels
+            beta  (float): weight of false negative voxels
+            smooth (float): epsilon to avoid division by zero, when both Numerator and Denominator of Tversky are zeros
+
+        Notes:
+            - setting alpha=beta=0.5: equivalent to DiceLoss
+            - default parameters were suggested by https://arxiv.org/pdf/1706.05721.pdf
+        """
+        super(TverskyLoss, self).__init__()
+        self.alpha = alpha
+        self.beta = beta
+        self.smooth = smooth
+
+    def tversky_index(self, y_pred, y_true):
+        """Compute Tversky index
+
+        Args:
+            y_pred (torch Tensor): prediction
+            y_true (torch Tensor): target
+
+        Returns:
+            float: Tversky index
+        """
+        # Compute TP
+        tp = torch.sum(y_true * y_pred)
+        # Compute FN
+        fn = torch.sum(y_true * (1 - y_pred))
+        # Compute FP
+        fp = torch.sum((1 - y_true) * y_pred)
+        # Compute Tversky for the current class, see Equation 3 of the original paper
+        numerator = tp + self.smooth
+        denominator = tp + self.alpha * fn + self.beta * fp + self.smooth
```

Comment:
It's a detail, but I think the alpha and beta are switched up, from the paper : 
> where α and β control the magnitude of penalties for FPs and FNs, respectively.

Reply:
The confusion probably comes from `eq3` since they define :
> the `p0i` is the probability of voxel `i` be a lesion and `p1i` is the probability of voxel `i` be a non-lesion

Which is kind of the opposite of what I would have expected

Reply:
Mmm this is confusing I agree. I ve just swapped!